### DISCUSSION TITLE
Forward declare CURL.

### DISCUSSION THREAD

**Discussion by hiic** [curl/curl#13828] [Q&A] [answered]
How do I forward declare CURL (C++)? Simply doing:
`class CURL;` gives `CURL redefinition; different basic types` error.

if I can’t forward declare I will need to include curl.hpp wich means any client including my header will need to include curl path to their includes path and the client has no direct dependency on curl.

thanks!

**Comment by dfandrich** [comment, marked as answer]
The curl project doesn't ship a curl.hpp file. It sounds like you're using a
curl language binding of some sort, so any questions about that binding should
be directed to its authors.
